### PR TITLE
Settings and initial contents for French localization

### PR DESCRIPTION
**Pull request by fydrah**
This PR is to initiate French localization based on:
* Issue #915
* https://github.com/cncf/glossary/blob/main/LOCALIZATION.md Guide

Base and target branch of this PR is `dev-fr` which is development branch for French localization.

This PR includes:
* `[languages.fr]` section in `config.toml`
* `content/fr/` directory
* `i18n/fr.toml`
* Some translated documents:
    * Main page: `content/fr/_index.md`
    * contribute: `content/fr/contribute/_index.md`
    * style-guide: `content/fr/style-guide/_index.md`
* Others:
    * FR Template: `content/fr/_TEMPLATE.md`

Since French localization team owns content/fr/* only,
this PR will require approvals from upstream owners (maintainers).

### CODE REVIEW

**Comment by seokho-son** [comment]
@fydrah! Thank you for the contribution ! :)

We are going to check following tasks in this PR. (target branch is dev-fr)

- [ ] 3-1. Adding a new language setting to site configuration https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-1-adding-a-new-language-setting-to-site-configuration
- [ ] 3-2. Adding a file for site strings https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-2-adding-a-file-for-site-strings
- [ ] 3-3. Adding a new localization directory https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-3-adding-a-new-localization-directory
- [ ] 3-4. Minimum required content https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-4-minimum-required-content
- [ ] 3-5. Check configuration works https://github.com/cncf/glossary/blob/main/LOCALIZATION.md#3-5-check-configuration-works-and-open-a-pr

In the meantime, 
1) Since maintainers cannot review French contents, could you invite volunteers who can review French contents in this PR ? (I expect two reviewers' LGTM)
2) Cloud you test Hugo Build locally for this PR to verify the configuration works ? (This page does not have Netlify preview yet)

**Comment by fydrah** [comment]
For french content, @Krast76 and @huats could you add your review?

**Comment by fydrah** [comment]
Hugo build ok and we can switch to french localization on the dropdown:
* Dropdown:
Screenshot from 2022-08-01 08-27-45
* switched to fr lang:
Screenshot from 2022-08-01 08-28-39

File: config.toml [hunk 1]
```diff
@@ -117,6 +117,13 @@ languageName = "বাংলা(Bengali)"
 contentDir = "content/bn"
 weight = 8
 
+[languages.fr]
+title = "Glossaire Cloud Native"
+description = "Le projet de Glossaire CNCF Cloud Natif a pour objectif de référencer les termes courants utilisés dans les différentes applications cloud native."
+languageName ="Français (French)"
```
Comment (seokho-son) [inline config.toml:123]:
```suggestion
languageName ="Français(French)"
```

For a consistency with other languages. (language selection UI is not long)

Comment (seokho-son) [reply config.toml:123]: Hi @fydrah
Note: I will add a new commit to apply my suggestion. :)
Actually, this is an intentional commit to initiate a Hugo build by Netlify. (we applied Netlify to dev-fr recently)
Netlify will provide a live preview site. :)

**Comment by seokho-son** [comment]
https://deploy-preview-1145--cncfglossary.netlify.app/fr/

:)

**Comment by seokho-son** [comment]
Now, all I need from this PR is LGTMs from Fr reviewers :)

File: i18n/fr.toml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+[post_create_child_page]
+other = "Créer une page enfant"
+[post_create_issue]
+other = "Ouvrir un ticket"
+[post_create_project_issue]
+other = "Ouvrir un ticket de projet"
+[post_posts_in]
+other = "Postés dans"
+[post_reading_time]
+other = "temps de lecture"
```
Comment (huats) [inline i18n/fr.toml:53]: "minutes de lecture" a la place ?

**Review by huats — COMMENTED** [review]
Je suis hésitant sur un. J'ai fais une proposition. J'attends vos retours

File: i18n/fr.toml (hunk 2 above)
Comment (Krast76) [reply i18n/fr.toml:53]: J'ai une préférence pour "temps de lecture" perso. A la limite si on veut préciser l'unité "temps de lecture (minutes)"

Comment (fydrah) [reply i18n/fr.toml:53]: Pour plus de context, cf. https://github.com/cncf/glossary/blob/main/i18n/en.toml#L53

En effet, on parlerait plus de `minute(s) de lecture` plutot que de "temps de lecture", comme dit Christophe son utilisation serait plus dans le sens `XY minute(s) de lecture` (et c'est le cas pour les autres traductions aussi).
Je modifie ca.

**Review by huats — APPROVED** [review]

**Comment by huats** [comment]
LGTM !

**Review by seokho-son — APPROVED** [review]
As a maintainer, LGTM :)

**Pull request merged**